emission/individual_tests/TestNominatim.py

```diff
@@ -0,0 +1,113 @@
+
+    #Testing get_json_geo, which passes in an address as a query. Compares three select k,v pairs in the results. 
+    def test_get_json_geo(self):
+        expected_result = {'place_id': 132490, 'licence': 'Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright', 'osm_type': 'way', 'osm_id': 141567710, 'boundingbox': ['41.8325787', '41.8332278', '-71.4161848', '-71.4152064'], 'lat': '41.8330097', 'lon': '-71.41568124868104', 'display_name': 'State of Rhode Island Department of Administration, 1, Park Street, Downtown, Providence, Providence County, 02908, United States', 'class': 'building', 'type': 'civic', 'importance': 1.75001}
+        actual_result = eco.Geocoder.get_json_geo("State of Rhode Island Department of Administration, 1, Park Street, Downtown, Providence, Providence County, 02908, United States")[0]
+        key_list = ['osm_id', 'boundingbox', 'display_name']
+        for k in key_list:
+            self.assertEqual(expected_result[k], actual_result[k])
+
+    #Testing the geocode function, which passes in an address and gets latitude and longitude.
+    # Test creates instance of coordinates using coordinate class. Getting lat and lon of the coordinate using get_lat and get_lon methods from the class.
+    def test_geocode(self):
+        expected_result_lon = Coordinate(41.8239891, -71.4128343).get_lon()
+        expected_result_lat = Coordinate(41.8239891, -71.4128343).get_lat()
+        actual_result = eco.Geocoder.geocode("Providence, Rhode Island")
+        actual_result_lon = actual_result.get_lon()
+        actual_result_lat = actual_result.get_lat()
+        self.assertEqual(expected_result_lon, actual_result_lon)
+        self.assertEqual(expected_result_lat, actual_result_lat)
+
+    #Testing make_url_reverse, which creates a query url from a lat and lon
+    def test_make_url_reverse(self):
+        expected_result = NOMINATIM_QUERY_URL + "/reverse?lat=41.8239891&lon=-71.4128343&format=json"
+        actual_result = (eco.Geocoder.make_url_reverse(41.8239891, -71.4128343))
+        self.assertEqual(expected_result, actual_result)
+ 
+    #Testing get_json_reverse, which reverse geocodes from a lat and lon. Tested result was modified to only look at the name returned with the coordinates, rather than the entire dictionary.
+    def test_get_json_reverse(self):
+        expected_result = "Providence City Hall"
+        actual_result = eco.Geocoder.get_json_reverse(41.8239891, -71.4128343)["display_name"].split(",")[0]
+        self.assertEqual(expected_result, actual_result)
+
+    #Testing reverse_geocode, which reverse geocodes from a lat and lon and returns only the display name. 
+    def test_reverse_geocode(self):
+        expected_result = "Portugal Parkway, Fox Point, Providence, Providence County, 02906, United States"
+        actual_result = eco.Geocoder.reverse_geocode(41.8174476, -71.3903767)
+        self.assertEqual(expected_result, actual_result)
```

Comment:
which of `get_json_reverse` and `reverse_geocode` is actually used in the current codebase? It seems like we should be able to remove the other. Doesn't hurt to have both, but simpler interfaces tend to be easier to maintain.

Reply:
Get_json_reverse is used in clean_and_resample's get_filtered_place function. 

reverse_geocode is used in emission/analysis/modelling/tour_model/tour_model_matrix.py and emission/incomplete_tests/TestProfile.py. 

However, reverse_geocode as it is currently in the code returns `_do_google_reverse(lat, lng)` (which I removed, along with all other google related functions), which used PyGeo.

Reply:
To give better context: reverse_geocode currently looks like this in production:
```
  @classmethod
    def reverse_geocode(cls, lat, lng):
        # try:
        #     jsn = cls.get_json_reverse(lat, lng)
        #     address = jsn["display_name"]
        #     return address

        # except:
        #     print "defaulting"
        return _do_google_reverse(lat, lng) # Just in case
```
Which uses this function that I also removed:
```
def _do_google_reverse(lat, lng):
    geo = pyGeo(GOOGLE_MAPS_KEY)
    address = geo.reverse_geocode(lat, lng)
    return address[0]
```
And to replace it, I just un-commented the portion out that used nominatim, making it look like this:
```
    @classmethod
    def reverse_geocode(cls, lat, lng):
        jsn = cls.get_json_reverse(lat, lng)
        address = jsn["display_name"]
        return address
```

Reply:
It seems like you can replace calls to `reverse_geocode` with `get_json_reverse.get("display_name")` and simplify the interface. But this is not critical for this PR - just a nice-to-have cleanup if you have time and motivation.